scheds/rust/scx_rustland/src/bpf/main.bpf.c

```diff
@@ -322,6 +322,12 @@ static bool is_task_cpu_available(struct task_struct *p)
 	if (is_kthread(p) && p->nr_cpus_allowed == 1)
 		return true;
 
+	/*
+	 * No scheduling required if it's the last task running.
+	 */
+        if (enq_flags & SCX_ENQ_LAST)
+		return true;
```

Comment:
I don't quite understand what difference this commit makes given that the default behavior w/o SCX_ENQ_LAST is auto-enq on local dsq. Does this make any behavior difference?

Reply:
> I don't quite understand what difference this commit makes given that the default behavior w/o SCX_ENQ_LAST is auto-enq on local dsq. Does this make any behavior difference?

For some reasons with this in place the CPU utilization of scx_rustland drops significantly when the system is idle (like 0.3-0.5%), without this change when the system is idle I can see some cpu usage spikes of up to 5-10%,but I'm not sure why it's happening... My theory was that with SCX_ENQ_LAST in place we could save some unnecessary usersched invocations.

Reply:
That's surprising given that the two essentially are doing the same thing. Maybe there's something timing dependent going on or I'm just confused.

Reply:
> That's surprising given that the two essentially are doing the same thing. Maybe there's something timing dependent going on or I'm just confused.

I'll run some tests with and without that and collect more info, now I want to understand what's going on exactly :)

Reply:
@htejun it's SCX_OPS_ENQ_LAST set in sched_ext_ops.flags that seems to make a difference, the check in `is_task_cpu_available()` does't make any difference and can be dropped apparently.